src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -897,6 +897,9 @@ ep_buffer_manager_write_event (
 	// Before we pick a buffer, make sure the event is enabled.
 	ep_return_false_if_nok (ep_event_is_enabled (ep_event));
 
+	// Check that the payload size is less than 64 KB (max size for ETW events)
+	ep_return_false_if_nok (ep_event_payload_get_size (payload) <= 64 * 1024);
```

Comment:
I'd recommend we add some debugging aids to help us and users understand when this might be happening:
1. Lets increment the sequence number so the missing events are detectable in the trace
2. Lets add a field to the buffer manager and increment it when this happens. That will let us look at any dump and be able to determine if any large events were dropped.